scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -941,10 +967,17 @@ class FormatWriter(formatOps: FormatOps) {
             text: Scaladoc.Text,
             termIndent: String
         ): Unit = {
-          // remove space as iterWords adds it
-          sb.setLength(sb.length - 1)
+          def prefixFirstWord(word: String): String = {
+            def checkFence(head: Char) =
+              (head == '`' || head == '~') && word.indexWhere(_ != head, 1) < 0
+            if (word.length >= 3 && checkFence(word(0)))
```

Comment:
Maybe just `if (word.startsWith("```") || word.startsWith("~~~"))`?

Reply:
i did it because the fence should consist of all characters being the same.

but now i am thinking that you might be right; we are trying to prevent a text to be _interpreted_ as a fence, and text like `~~~scala` will definitely be parsed into fence and info string.